src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -1614,12 +1662,13 @@ private bool DoMatchWorker(string operandString, MatchInfo matchInfo, out MatchI
                 }
 
                 // otherwise construct and populate a new MatchInfo object
-                matchResult = new MatchInfo
-                              {
-                                  IgnoreCase = !CaseSensitive,
-                                  Line = operandString,
-                                  Pattern = Pattern[patternIndex]
-                              };
+                matchResult = new MatchInfo(Raw, true, 0, 0);
+                              
+                matchResult.IgnoreCase = !CaseSensitive;
```

Comment:
Please use the [object initialiser notation](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/how-to-initialize-objects-by-using-an-object-initializer)